keanu-project/src/main/java/io/improbable/keanu/vertices/dbltensor/Nd4jDoubleTensor.java

```diff
@@ -0,0 +1,430 @@
+    public DoubleTensor powInPlace(DoubleTensor exponent) {
+        if (exponent.isScalar()) {
+            Transforms.pow(tensor, exponent.scalar(), false);
+        } else {
+            INDArray exponentArray = unsafeGetNd4J(exponent);
+            Transforms.pow(tensor, exponentArray, false);
+        }
+        return this;
+    }
+
+    @Override
+    public DoubleTensor powInPlace(double exponent) {
+        Transforms.pow(tensor, exponent, false);
+        return this;
+    }
+
+    @Override
+    public DoubleTensor logInPlace() {
+        Transforms.log(tensor, false);
+        return this;
+    }
+
+    @Override
+    public DoubleTensor sinInPlace() {
+        Transforms.sin(tensor, false);
+        return this;
+    }
+
+    @Override
+    public DoubleTensor cosInPlace() {
+        Transforms.cos(tensor, false);
+        return this;
+    }
+
+    @Override
+    public DoubleTensor asinInPlace() {
+        Transforms.asin(tensor, false);
+        return this;
+    }
+
+    @Override
+    public DoubleTensor acosInPlace() {
+        Transforms.acos(tensor, false);
+        return this;
+    }
+
+    @Override
+    public DoubleTensor expInPlace() {
+        Transforms.exp(tensor, false);
+        return this;
+    }
+
+    @Override
+    public DoubleTensor minusInPlace(DoubleTensor that) {
+        if (that.isScalar()) {
+            tensor.subi(that.scalar());
+        } else {
+            INDArray thatArray = unsafeGetNd4J(that);
+            tensor.subi(thatArray);
+        }
+        return this;
+    }
+
+    @Override
+    public DoubleTensor plusInPlace(DoubleTensor that) {
+        if (that.isScalar()) {
+            tensor.addi(that.scalar());
+        } else {
+            INDArray thatArray = unsafeGetNd4J(that);
+            tensor.addi(thatArray);
+        }
+        return this;
+    }
+
+    @Override
+    public DoubleTensor timesInPlace(DoubleTensor that) {
+        if (that.isScalar()) {
+            tensor.muli(that.scalar());
+        } else {
+            INDArray thatArray = unsafeGetNd4J(that);
+            tensor.muli(thatArray);
+        }
+        return this;
+    }
+
+    @Override
+    public DoubleTensor divInPlace(DoubleTensor that) {
+        if (that.isScalar()) {
+            tensor.divi(that.scalar());
+        } else {
+            INDArray thatArray = unsafeGetNd4J(that);
+            tensor.divi(thatArray);
+        }
+        return this;
+    }
+
+    @Override
+    public DoubleTensor unaryMinusInPlace() {
+        tensor.negi();
+        return this;
+    }
+
+    @Override
+    public DoubleTensor getGreaterThanMask(DoubleTensor greaterThanThis) {
+        INDArray greaterThanThisArray = unsafeGetNd4J(greaterThanThis);
+        INDArray mask = tensor.dup();
+
+        if (greaterThanThisArray.isScalar()) {
+            Nd4j.getExecutioner().exec(
+                    new OldGreaterThan(mask, Nd4j.ones(mask.shape()).mul(greaterThanThisArray.getDouble(0)), mask, mask.length())
```

Comment:
Given we're in this conditional due to it being scalar, can we call `greaterThanThisArray.getScalar(0)`? Looks a bit clearer

Reply:
Great spot!